spyder/plugins/editor/plugin.py

```diff
@@ -1353,12 +1353,16 @@ def register_plugin(self):
         self.add_dockwidget()
 
         # Add modes to switcher
-        self.switcher_manager = EditorSwitcherManager(
-            self,
-            self.main.switcher,
-            self.get_current_editor,
-            self.get_current_editorstack,
-            section=self.get_plugin_title())
+        # TODO: 'Switcher' object has no attribute 'add_mode' 
+        # it is needed to create a public API that contains the methods 
+        # that handles the EditorSwitcherManager
+
+        # self.switcher_manager = EditorSwitcherManager(
+        #     self,
+        #     self.main.switcher,
+        #     self.get_current_editor,
+        #     self.get_current_editorstack,
+        #     section=self.get_plugin_title())
```

Comment:
Checked and we could keep the switcher working by uncomenting this and the import for the `EditorSwitcherManager` if we also change the second `arg` from `self.main.switcher` to be `self.main.switcher.get_container().switcher`:

```suggestion
        self.switcher_manager = EditorSwitcherManager(
            self,
            self.main.switcher.get_container().switcher,
            self.get_current_editor,
            self.get_current_editorstack,
            section=self.get_plugin_title())
```